vfs/http_streambuf.cc

```diff
@@ -87,6 +87,14 @@ struct HttpStreamingDownloadSource {
         return future.get();
     }
 
+    shared_ptr<FsObjectInfo> getObjectInfo() const
```

Comment:
I told you you would cry... This object is available to the caller so this is why I picked it.